concordium-consensus/src/Concordium/KonsensusV1/TreeState/Implementation.hs

```diff
@@ -774,16 +785,20 @@ finalizeTransactions = mapM_ removeTrans
     removeTrans WithMetadata{wmdData = NormalTransaction tr, ..} = do
         let nonce = transactionNonce tr
             sender = accountAddressEmbed (transactionSender tr)
-        anft <- use (transactionTable . TT.ttNonFinalizedTransactions . at' sender . non TT.emptyANFT)
-        unless (anft ^. TT.anftNextNonce == nonce) $
+        senderMinNonce <- do
+            macct <- flip getAccount (aaeAddress sender) =<< use (lastFinalized . to bpState)
+            fromMaybe minNonce <$> mapM (getAccountNonce . snd) macct
+        nextNonce <- use (transactionTable . TT.ttNonFinalizedTransactions . at' sender . non (TT.emptyANFTWithNonce senderMinNonce) . TT.anftNextNonce)
+        unless (nextNonce == nonce) $
```

Comment:
I think it's a bad idea to look up the account for each transaction we finalize here. The `unless` check is a debug assertion, it is not meant to be triggered.

Also, since you are finalizing transactions, should they not be in the `ttNonFinalizedTransactions` already?